pkgs/applications/emulators/dolphin-emu/primehack.nix

```diff
@@ -1,138 +1,164 @@
 , cmake
+, pkg-config
 , wrapQtAppsHook
-, qtbase
+, alsa-lib
 , bluez
+, bzip2
+, cubeb
+, curl
+, enet
 , ffmpeg
-, libao
-, libGLU
+, fmt_10
+, gtest
+, hidapi
+, libevdev
 , libGL
-, pcre
-, gettext
-, libXrandr
+, libiconv
+, libpulseaudio
+, libspng
 , libusb1
-, libpthreadstubs
-, libXext
-, libXxf86vm
-, libXinerama
-, libSM
 , libXdmcp
-, readline
-, openal
-, udev
-, libevdev
-, portaudio
-, curl
-, alsa-lib
-, miniupnpc
-, enet
+, libXext
+, libXrandr
+, lz4
+, lzo
 , mbedtls_2
-, soundtouch
+, miniupnpc
+, openal
+, pugixml
+, qtbase
+, qtsvg
 , sfml
-, fmt
-, xz
+, udev
 , vulkan-loader
-, libpulseaudio
+, xxHash
+, xz
 
-# - Inputs used for Darwin
+  # Darwin-only dependencies
 , CoreBluetooth
 , ForceFeedback
+, IOBluetooth
 , IOKit
+, moltenvk
 , OpenGL
-, libpng
-, hidapi
+, VideoToolbox
 }:
 
 stdenv.mkDerivation rec {
   pname = "dolphin-emu-primehack";
-  version = "1.0.6a";
+  version = "1.0.7a";
 
   src = fetchFromGitHub {
     owner = "shiiion";
     repo = "dolphin";
     rev = version;
-    sha256 = "sha256-gc4+ofoLKR+cvm+SaWEnGaKrSjWMKq7pF6pEIi75Rtk=";
+    sha256 = "sha256-vuTSXQHnR4HxAGGiPg5tUzfiXROU3+E9kyjH+T6zVmc=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # fix build w/ glibc-2.39
+    (fetchpatch {
+      url = "https://github.com/dolphin-emu/dolphin/commit/3da2e15e6b95f02f66df461e87c8b896e450fdab.patch";
+      hash = "sha256-+8yGF412wQUYbyEuYWd41pgOgEbhCaezexxcI5CNehc=";
+    })
+  ];
+
   nativeBuildInputs = [
-    pkg-config
+    stdenv.cc
     cmake
-  ] ++ lib.optional stdenv.isLinux wrapQtAppsHook;
+    pkg-config
+    wrapQtAppsHook
+  ];
 
-  buildInputs = [
+  buildInputs = lib.optionals stdenv.isDarwin [
+    CoreBluetooth
+    ForceFeedback
+    IOBluetooth
+    IOKit
+    moltenvk
+    OpenGL
+    VideoToolbox
+  ] ++ [
+    bzip2
+    cubeb
     curl
+    enet
     ffmpeg
-    libao
-    libGLU
-    libGL
-    pcre
-    gettext
-    libpthreadstubs
+    fmt_10
+    gtest
+    hidapi
+    libiconv
     libpulseaudio
-    libXrandr
-    libXext
-    libXxf86vm
-    libXinerama
-    libSM
-    readline
-    openal
-    libXdmcp
-    portaudio
+    libspng
     libusb1
-    libpng
-    hidapi
-    miniupnpc
-    enet
+    libXdmcp
+    lz4
+    lzo
     mbedtls_2
-    soundtouch
-    sfml
-    fmt
-    xz
+    miniupnpc
+    #minizip-ng # Not wanting to make a mess with patches, leaving the vendored one for now
+    openal
+    pugixml
     qtbase
+    qtsvg
+    #SDL2 # Not used yet
+    sfml
+    xxHash
+    xz # LibLZMA
+    # Causes linker errors with minizip-ng, prefer vendored. Possible reason why: https://github.com/dolphin-emu/dolphin/pull/12070#issuecomment-1677311838
+    #zlib-ng
   ] ++ lib.optionals stdenv.isLinux [
+    alsa-lib
     bluez
-    udev
     libevdev
-    alsa-lib
+    libGL
+    libXext
+    libXrandr
+    # FIXME: Vendored version is newer than mgba's stable release, remove the comment on next mgba's version
+    #mgba # Derivation doesn't support Darwin
```

Comment:
We could use an overlay

Reply:
[Oh, no...](https://github.com/NixOS/nixpkgs/pull/309064#discussion_r1595249685)